src/core/util.js

```diff
@@ -283,4 +285,221 @@ var parseTimeSeriesForC3 = function(graph_data) {
+        var lastIndex = 0;
+        while (year_range_re.test(model)){
+            year_range_re.lastIndex = lastIndex;
+            year_range.push(year_range_re.exec(model)[0].slice(0,4));
+            lastIndex = year_range_re.lastIndex;
+        }
+        var period = year_range[0] + " - " + year_range[1];
+        var modelInfo = {
+            "model_period": period,
+            "run": data[model]['run'],
+            "min": +data[model]['min'].toFixed(PRECISION),
+            "max": +data[model]['max'].toFixed(PRECISION),
+            "w_mean": +data[model]['mean'].toFixed(PRECISION),
+            "median": +data[model]['median'].toFixed(PRECISION),
+            "w_stdev": +data[model]['stdev'].toFixed(PRECISION),
+            "units": data[model]['units']
+        };
+        flatData.push(modelInfo); 
+    }
+    return flatData;
+}
+
+/*
+    Helper function for exportTableDataToWorksheet, creates summary rows that appear at the top of the exported worksheet
+*/
+var createWorksheetSummaryCells = function(summary_data, time_of_year) {
+    // store worksheet cell contents to be later encoded as per output format
+    var cells = [];
+    var cell;
+    // write summary rows at top of worksheet
+    var summary_header = ["Model", "Emissions Scenario", "Time of Year", "Variable ID", "Variable Name"];
+    var summary_keys = ["model_id", "experiment", "time_of_year", "variable_id", "variable_name"];
+
+    var num_summary_rows = 3;
+    var num_summary_cols = summary_keys.length
+    for(var R = 0; R < num_summary_rows; ++R) {
+        for(var C = 0; C < num_summary_cols; ++C) {
+            if(R == 0) cell = {v: summary_header[C]};
+            else if(R == 1) {
+                if(summary_keys[C] == 'time_of_year') {
+                  cell = {v: time_of_year};      
+                }
+                else cell = {v: summary_data[summary_keys[C]]};
+            }
+            else cell = {v: ""};
+            cell.t = 's';
+            cells.push(cell);
+        }
+    }
+    return { 'num_rows': num_summary_rows, 'num_cols': num_summary_cols, 'cells': cells };
+}
+
+/*
+    Helper function for exportTableDataToWorksheet, creates data column headers and data entries for exported worksheet
+*/
+var fillWorksheetDataCells = function(data) {
+    // store worksheet cell contents to be later encoded as per output format
+    var cells = [];
+    var cell;
+    // populate worksheet with table data
+    var column_labels = ["Model Period", "Run", "Min", "Max", "W.Mean", "Median", "W.Std.Dev", "Units" ];
+    var data_keys = ["model_period", "run", "min", "max", "w_mean", "median", "w_stdev", "units" ];
+    var num_data_rows = Object.keys(data).length + 1;
+    var num_data_cols = data_keys.length
+
+    for(var R = 0; R < num_data_rows; ++R) {
+        for(var C = 0; C < num_data_cols; ++C) {
+            // create header row
+            if(R == 0) cell = {v: column_labels[C]};
+            // create cell object: .v is the actual data
```

Comment:
And... what is `.t`? Maybe some notes in the function header about what is the return value of this function, because it's not overtly apparent.